runtime/actors/agents.go

```diff
@@ -246,7 +256,7 @@ func PublishAgentEvent(ctx context.Context, agentId, eventName string, eventData
 	event := &messages.AgentEventMessage{
 		Name:      eventName,
 		Data:      dataValue,
-		Timestamp: timestamppb.Now(),
```

Comment:
The problem stems from this `Timestamp: timestamppb.Now()` - its the published time, not the captured time.